### PR TITLE
Use Merge with AppendValues option when merging default config with hints generated config

### PR DESCRIPTION
**Pull request by MichaelKatsoulis**
This PR is the resolution of https://github.com/elastic/beats/issues/36838 issue where the problem is described in details.

In filebeat, when kubernetes autodisocver is used with hints enabled, processors and other array fields (paths, parsers) are not merged correctly.
The values coming from default config and the ones generated from hints should be merged by appending the hints generated values to the default config's ones in case of arrays like processors.

The `Merge` method used does not append array values by default. 
Instead `MergeWithOpts` and `AppendValues` option should be used.

### CODE REVIEW

File: filebeat/autodiscover/builder/hints/logs.go [hunk 1]
```diff
@@ -139,7 +139,8 @@ func (l *logHints) CreateConfig(event bus.Event, options ...ucfg.Option) []*conf
 			kubernetes.ShouldPut(tempCfg, json, jsonOpts, l.log)
 		}
 		// Merge config template with the configs from the annotations
```
Comment (gizas) [inline filebeat/autodiscover/builder/hints/logs.go:141]: Remove this now. Is not needed probably

**Review by gizas — APPROVED** [review]

File: filebeat/autodiscover/builder/hints/logs.go (hunk 1 above)
Comment (MichaelKatsoulis) [reply filebeat/autodiscover/builder/hints/logs.go:141]: I think it is still a valid comment as it describes what is merged. And then we describe why AppendValues option is used.

**Review by ChrsMark — APPROVED** [review]
LGTM. I think a changelog entry would be needed here since it fixes a user facing issue.

**Review by ChrsMark — APPROVED** [review]

**Comment by MichaelKatsoulis** [comment]
@elastic/elastic-agent-data-plane could you please review this PR as you are the code owners ?

**Review by cmacknz — APPROVED** [review]

**Pull request merged**